File: docs/guides/javascript/src/searchWithRuleContexts.ts
import { algoliasearch } from 'algoliasearch';

import type { SearchParams } from 'algoliasearch';

const getPlatformTag = () => {
  return ''; // Implement your logic here
};

const client = algoliasearch('ALGOLIA_APPLICATION_ID', 'ALGOLIA_API_KEY');

const platformTag = getPlatformTag();
const searchParams: SearchParams = {
  query: '<YOUR_SEARCH_QUERY>',
  ruleContexts: [platformTag],
};

await client.searchSingleIndex({ indexName: 'indexName', searchParams: searchParams });

File: docs/guides/python/searchWithRuleContexts.py
from algoliasearch.search.client import SearchClientSync

from algoliasearch.search.models.search_params import SearchParams


def _get_platform_tag() -> str:
    # Implement your logic here
    return ""


_client = SearchClientSync("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")


# get the buyer account information
platform_tag = _get_platform_tag()
search_params = SearchParams(
    query="<YOUR_SEARCH_QUERY>",
    rule_contexts=[platform_tag],
)

_client.search_single_index(
    index_name="<YOUR_INDEX_NAME>",
    search_params=search_params,
)
